testing/include/gz/common/testing/CMakeTestPaths.hh

```diff
@@ -28,21 +28,34 @@ namespace gz::common::testing
 ///
 /// It is not intended that users will directly construct this, but rather
 /// utilize the TestPathFactory.
+
+/// TODO(jrivero) Remove warning disable when bumping MSVC on CI
+/// MSVC v16.11.25 complains about the TestPaths class not having a DLL although
+/// relevant members of it have visiblity defined:
+///   "non dll-interface class 'gz::common::testing::TestPaths' used as base for
+///   dll-interface class 'gz::common::testing::CMakeTestPaths'"
+///
+/// Workaround on this problems triggers new warnings and suffering and
+/// more recent versions of MSVC do not trigger that warning anymore.
+/// Conclusion: disable the warning.
+#ifdef _MSC_VER
+#pragma warning(push)
+#pragma warning(disable: 4275)
+#endif
 class GZ_COMMON_TESTING_VISIBLE CMakeTestPaths: public TestPaths
 {
   /// \brief Constructor from TestPaths
   public: using TestPaths::TestPaths;
 
   /// \brief Destructor
-  public: GZ_COMMON_TESTING_VISIBLE ~CMakeTestPaths() override;
+  public: ~CMakeTestPaths() override;
 
   /// Documentation inherited
-  public: bool GZ_COMMON_TESTING_VISIBLE
-          ProjectSourcePath(std::string &_sourceDir) override;
+  public: bool ProjectSourcePath(std::string &_sourceDir) override;
 
   /// Documentation inherited
-  public: bool GZ_COMMON_TESTING_VISIBLE
-          TestTmpPath(std::string &_tmpDir) override;
+  public: bool TestTmpPath(std::string &_tmpDir) override;
 };
 }  // namespace gz::common::testing
+# pragma warning( pop )
```

Comment:
Oh, this needs to be wrapped in _MSC_VER as well

Reply:
[c604202](https://github.com/gazebosim/gz-common/pull/550/commits/c60420297a1cb3e0a5a3d91fd64e5b784fff6f80)